ckanext/datagovuk/plugin.py

```diff
@@ -257,10 +253,15 @@ def dgu_home():
         # also monkeypatch occurrence in original module as some views
         # call it directly instead of redirecting externally
         import ckan.views.user as ckan_user_views
+        ckan_user_views._edit_form_to_db_schema = user_edit_form_schema
+        ckan_user_views._new_form_to_db_schema = user_new_form_schema
         ckan_user_views.me = me
 
         return bp
 
+    # import these for monkey patching
+    from ckanext.datagovuk.ckan_patches import cli, logic, query
```

Comment:
It feels dangerous to have the monkey patching happen as a side effect of an import. Do you actually need `logic` and `query` here too?

Was there something wrong with the old approach? Or could we have a separate function on `ckan_patches` which does the monkey patching like:

```
from ckanext.datagovuk.ckan_patches.cli import monkey_patch
monkey_patch()
```

Reply:
The old approach included these monkey patches. Logic is actually the one that handles the user schema, query blocks some search queries which were causing problems in the past. 
This has been fixed by upgrading the service affected but it's probably a good idea to continue blocking the tags.

Reply:
Before this PR, `logic` and `query` weren't being imported, or am I missing something? Are you saying that they used to be imported, before the upgrade, but they got removed at some point? So now we're adding them back?

I see the approach of monkey patching as a side effect of an import has been around for the last 5 years (in `logic.py`), so I guess I'm fine with it for this PR. It's a big code smell though - imports shouldn't have side effects, and they doubly shouldn't have monkey patching side effects.

Reply:
Pre 2.9 they were being imported, it was removed in this commit https://github.com/alphagov/ckanext-datagovuk/commit/270bc2da195cbc46ca38f2afcc0eec1d44da7647#diff-b5ec20d532632efd2db909f73caa5ff88a59ab0a1664dd6ceb3a8f4a31ae4df9 probably by accident as that commit was part of the 2.9 upgrade branch. 
I think the problem was that there weren't any tests in place to check that it was allowing through more relaxed user names as the user schema in DGU is much more relaxed, this test has now been added in to prevent this from happening in the future.